entrypoint.sh

```diff
@@ -46,8 +46,19 @@ if [ "$1" = "kubelet" ]; then
         ln -sf /host/usr/share/ros/os-release /usr/lib/os-release
     fi
 
+    # Check if no other or additional resolv-conf is passed (default is configured as /etc/resolv.conf)
+    if echo "$@" | grep -q -- --resolv-conf=/etc/resolv.conf; then
+        # Check if host is running `system-resolved`
+        if pgrep -f systemd-resolved > /dev/null; then
+            # Check if the resolv.conf with the actual nameservers is present
+            if [ -f /run/systemd/resolve/resolv.conf ]; then
+                RESOLVCONF="--resolv-conf=/run/systemd/resolve/resolv.conf"
```

Comment:
Nit: can you do the same thing like we did in `--cgroup-driver` and not include the --resolve-conf in the environment variable.

Reply:
cgroup-driver is always configured, while `resolv-conf` will only be added in the condition that `systemd-resolved` is running. If we do this, we need to default to `/etc/resolv.conf` in this script, and it will always be added again which breaks the current logic.